### PR TITLE
Issue 882 - some initial policy doc

### PR DESCRIPTION
**Pull request by dabooz**
Signed-off-by: Dave Booz <booz@us.ibm.com>

### CODE REVIEW

File: docs/policy.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Because deployment policies capture the more dynamic, business-like service properties and constraints, they are expected to change more often than service policy. Their lifecycle is independent from the service they refer to, which gives the policy administrator the ability to state a specific service version or a version range.
+The deployment engine merges service policy and deployment policy (by performing a logical AND of the 2 policies), and then attempts to find nodes whose policy is compatible with that merged policy.
+
+## Model policy
+
+Machine learning (ML)-based services require specific trained models to operate correctly.
+OpenHorizon provides the ability to deploy models independently from services.
+In ML use case, the model tends to be large and change more more often than the algorithmic code which uses the model to analyze data.
+Model policy enables he administrator to deploy specific models on the same, or a subset of, nodes where the services that use the mode have been placed.
```
Comment (linggao) [inline docs/policy.md:54]: enables the

File: docs/properties_and_constraints.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+OpenHorizon provides some `built-in` properties.
+There is documentation for the [built-in properties](./built_in_policy.md).
+
+## Properties
+
+Properties are statements of fact about a node, a service implementation, a model, a deployment, etc.
+They are most commonly attached to nodes, enabling constraints in a deployment policy to select the nodes where services should be deployed.
+
+Properties are simple name value pairs, e.g. model = red.
```
Comment (linggao) [inline docs/properties_and_constraints.md:21]: color = red.

File: docs/properties_and_constraints.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+},
+{
+	"name": "versionProperty",      /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "version",
+	"value": "1.2.7"
+},
+{
+	"name": "losProperty",          /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "list of strings",
+	"value": "value1,value2"
```
Comment (linggao) [inline docs/properties_and_constraints.md:70]: list of strings are expressed as `"value": ["abcd", "1234"]`

File: docs/properties_and_constraints.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+}
+```
+
+## Constraints
+
+Constraints are the other half of the policy expression language in OpenHorizon.
+Think of constraints as selection predicates used to select nodes based on the properties defined and set on those nodes.
+Constraints are most commonly found in deployment policies but can be specified on nodes, service and model policy.
```
Comment (linggao) [inline docs/properties_and_constraints.md:78]: on node, service and model policy.

File: docs/properties_and_constraints.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+For example, using the example properties defined in the properties section above, the constraint:
+`"booleanProperty = true AND floatProperty < 1.0"` will evaluate to false.
+
+Each property type has operators that can be used to evaluate property values:
+* `string` - the operators `==` or `=` denote equals to and `!=` denotes not equal to.
+* `int` - supports the operators `==, <, >, <=, >=, =, !=`.
+* `boolean` - supports `==, =`
+* `float` - supports the operators `==, <, >, <=, >=, =, !=`.
+* `version` - supports `==, =, in` where in is used to indicate that a version is within a given range, e.g. any version 1 service: "(1.0.0,2.0.0]".
```
Comment (linggao) [inline docs/properties_and_constraints.md:98]: any version 1 service?

File: docs/policy.md (hunk 1 above)
Comment (dabooz) [reply docs/policy.md:54]: fixed

File: docs/properties_and_constraints.md (hunk 2 above)
Comment (dabooz) [reply docs/properties_and_constraints.md:21]: fixed

File: docs/properties_and_constraints.md (hunk 3 above)
Comment (dabooz) [reply docs/properties_and_constraints.md:70]: yep, it is different

File: docs/properties_and_constraints.md (hunk 4 above)
Comment (dabooz) [reply docs/properties_and_constraints.md:78]: fixed

File: docs/properties_and_constraints.md (hunk 5 above)
Comment (dabooz) [reply docs/properties_and_constraints.md:98]: yes, isn't that the version range for any version 1 service?

Comment (linggao) [reply docs/properties_and_constraints.md:98]: You are right. I was not sure what you were talking about. Now I understand.

Comment (dabooz) [reply docs/properties_and_constraints.md:98]: fixed

**Review by linggao — APPROVED** [review]

**Pull request merged**